File: src/ChpokkWeb/SystemFiles/Templates/ProjectTemplates/CSharp/Silverlight/1042/BusinessApplication/Models/User.partial.cs
namespace $safeprojectname$.Web
{
    using System;
    using System.ComponentModel;

    /// <summary>
    /// <see cref="User"/> 클래스에 대한 확장입니다.
    /// </summary>
    public partial class User
    {
        #region DisplayName을 바인딩 가능으로 지정

        /// <summary>
        /// <see cref="User.DisplayName"/>이 변경될 때 속성 변경 알림을 생성하는 <c>OnPropertyChanged</c> 메서드의 재정의입니다.
        /// </summary>
        /// <param name="e">속성 변경 이벤트 인수입니다.</param>
        protected override void OnPropertyChanged(PropertyChangedEventArgs e)
        {
            base.OnPropertyChanged(e);

            if (e.PropertyName == "Name" || e.PropertyName == "FriendlyName")
            {
                this.RaisePropertyChanged("DisplayName");
            }
        }
        #endregion
    }
}

File: src/ChpokkWeb/SystemFiles/Templates/ProjectTemplates/VisualBasic/Silverlight/3082/BusinessApplication/Models/User.partial.vb
Imports System
Imports System.ComponentModel

Namespace Web

    Partial Public Class User
#Region "Convertir DisplayName en enlazable"

        ''' <summary>
        ''' Invalidación del método <c>OnPropertyChanged</c> que genera notificaciones de cambio de propiedades cuando cambia <see cref="User.DisplayName"/>.
        ''' </summary>
        ''' <param name="e">Argumentos del evento de cambio de la propiedad</param>
        Protected Overrides Sub OnPropertyChanged(ByVal e As System.ComponentModel.PropertyChangedEventArgs)
            MyBase.OnPropertyChanged(e)

            If e.PropertyName = "Name" Or e.PropertyName = "FriendlyName" Then
                Me.RaisePropertyChanged("DisplayName")
            End If
        End Sub
#End Region
    End Class
End Namespace